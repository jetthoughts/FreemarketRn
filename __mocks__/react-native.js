const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
