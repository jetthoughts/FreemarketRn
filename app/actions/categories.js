export const loadCategories = () => ({ type: 'LOAD_CATEGORIES' });
export const selectCategory = (id) => ({ type: 'SELECT_CATEGORY', id });
