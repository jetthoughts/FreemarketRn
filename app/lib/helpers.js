const padTime = number => (number < 10 ? '0' + number : number);

export const dateToTimeString = (date) => (
  [
    padTime(date.getHours()), 
    padTime(date.getMinutes())
  ].join(':')
);

export const categoriesToObject = records => {
  const object = {};
  records.forEach(record => object[record.id] = record.name);
  return object;
}