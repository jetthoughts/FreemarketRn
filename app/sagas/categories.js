import { put, call, takeLatest } from 'redux-saga/effects';

import {
  categoriesSuccess,
  categoriesError,
  requestCategories,
} from '../reducers/categories';
import { fetchAllCategories } from '../Api';

export function* fetchCategories() {
  try {
    const response = yield call(fetchAllCategories);
    const [, ...categories] = response.data;
    yield put(categoriesSuccess(categories));
  } catch (e) {
    yield put(categoriesError(e));
  }
}

export function* watchCategoriesRequest() {
  yield takeLatest(requestCategories.getType(), fetchCategories);
}
