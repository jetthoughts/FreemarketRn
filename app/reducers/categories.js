import { combineReducers } from 'redux';
import { createAction, createReducer } from 'redux-act';
import categories from '../constants/categories';

export const loadCategories = createAction('load categories list');
export const selectCategory = createAction('change current category');

const listReducer = createReducer({
  [loadCategories]: (state) => categories,
}, []);

const currentReducer = createReducer({
  [selectCategory]: (state, payload) => payload,
}, 0)

export default combineReducers({
  list: listReducer,
  current: currentReducer,
});
