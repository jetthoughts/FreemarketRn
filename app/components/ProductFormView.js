import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ScrollView, View, Button, StyleSheet } from 'react-native';
import t from 'tcomb-form-native';

import { dateToTimeString } from '../lib/helpers'
import MaskedInputTemplate from '../lib/tcomb-form/templates/MaskedInputTemplate';
import LocalImageFactory from '../lib/tcomb-form/factories/LocalImageFactory';
import LocalImage from '../lib/tcomb-form/types/localImage';

const styles = StyleSheet.create({
  formView: {
    padding: 10,
  },
  button: {
    paddingTop: 10,
    paddingBottom: 10,
  },
});

const Form = t.form.Form;

const Name = t.String;

Name.getValidationErrorMessage = (value, path, context) => {
  if (!value) {
    return "Please, add some name";
  } else {
    return '';
  }
}

const Positive = t.refinement(t.Number, function (n) {
  return n >= 0 && n <= 100;
});

Positive.getValidationErrorMessage = (value, path, context) => {
  if (value > 100) {
    return "Too expensive, nobody gonna buy it";
  } else {
    return 'Please, give a positive number';
  }
}

const productOptions = {
  image: LocalImage,
  name: Name,
  description: t.maybe(t.String),
  category: t.enums({'-1': 'Loading...'}),
  price: Positive,
  allowPhone: t.Bool,
  phone: t.String,
  endTime: t.Date,
};

const formOptions = {
  i18n: {
    optional: '',
    required: ' *',
  },
  fields: {
    image: {
      factory: LocalImageFactory,
      error: 'Please, add an image',
    },
    name: {
      maxLength: 10,
      error: 'Please, add some name',
    },
    description: {
      multiline: true,
      numberOfLines: 4,
    },
    category: {
      nullOption: {
        value: '', 
        text: '< Choose Category >',
      },
    },
    price: {
      help: 'Pro tip: better to set affordable price',
    },
    phone: {
      editable: false,
      template: MaskedInputTemplate,
      config: {
        mask: 'cel-phone',
      }
    },
    endTime: {
      mode: 'time',
      config: {
        format: dateToTimeString,
      },
    },
  },
};

export default class ProductFormView extends Component {
  constructor(props) {
    super(props);

    props.loadCategories();

    const newProductOptions = t.update(productOptions, {
      category: { '$set': t.enums(props.categories) }
    });
    
    this.state = {
      imageSrouce: null,
      localImage: null,
      formOptions: formOptions,
      formValue: {
        name: 'Misha',
      },
      productOptions: newProductOptions,
    };
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.categories != this.props.categories) {
      const newProductOptions = t.update(this.state.productOptions, {
        category: { '$set': t.enums(nextProps.categories) }
      });
      this.setState({
        productOptions: newProductOptions,
      });
    }
  }

  onFormPress() {
    const product = this.form.getValue();
    this.props.onPress({ product });
  }

  onFormChange(value) {
    const allowPhone = !!value.allowPhone;
    const options = t.update(this.state.formOptions, {
      fields: {
        phone: {
          editable: {'$set': allowPhone},
        }
      }
    });
    const newProductOptions = t.update(this.state.productOptions, {
      phone: {
        '$set': (allowPhone ? t.String : t.maybe(t.String)),
      },
    })
    this.setState({
      formOptions: options, 
      formValue: value, 
      productOptions: newProductOptions
    });
  }

  render() {
    return (
      <ScrollView style={styles.formView}>
        <Form
          ref={_form => (this.form = _form)}
          type={t.struct(this.state.productOptions)}
          options={this.state.formOptions}
          value={this.state.formValue}
          onChange={(value) => this.onFormChange(value)}
        />
        <View style={styles.button}>
          <Button
            onPress={() => this.onFormPress()}
            title="Add Product"
          />
        </View>
        <View style={styles.button}>
          <Button
            style={styles.button}
            onPress={() => this.props.onBackPress()}
            color="red"
            title="Back"
          />
        </View>
      </ScrollView>
    );
  }
}

ProductFormView.propTypes = {
  onPress: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
};
