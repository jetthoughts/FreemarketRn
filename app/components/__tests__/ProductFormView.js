import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MockDate from 'mockdate';

import ProductFormView from '../ProductFormView';
import categories from '../../constants/categories';
import { categoriesToObject } from '../../lib/helpers';

it('renders successfully', () => {
  MockDate.set('Wed Feb 02 2000 00:00:00', 120);

  const tree = renderer.create(
    <ProductFormView
      categories={categoriesToObject(categories)}
      onPress={() => true}
      onBackPress={() => true}
      loadCategories={() => true}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();

  MockDate.reset();
});
