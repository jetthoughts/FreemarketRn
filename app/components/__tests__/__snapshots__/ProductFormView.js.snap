// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<RCTScrollView
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "height": 150,
              "width": 150,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 1,
                "borderStyle": "dotted",
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Place for an Image
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
              "width": 150,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#0C42FD",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Select Image
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Name *
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Name *"
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={10}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value="Misha"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Description
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Description"
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={true}
            numberOfLines={4}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Category *
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                }
              }
            >
              &lt; Choose Category &gt;
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={Object {}}
            >
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "< Choose Category >",
                      "textColor": undefined,
                      "value": "",
                    },
                    Object {
                      "label": "Antiques",
                      "textColor": undefined,
                      "value": "0",
                    },
                    Object {
                      "label": "Books, Comics & Magazines",
                      "textColor": undefined,
                      "value": "1",
                    },
                    Object {
                      "label": "Cars, Motorcycles & Vehicles",
                      "textColor": undefined,
                      "value": "2",
                    },
                    Object {
                      "label": "Events Tickets",
                      "textColor": undefined,
                      "value": "3",
                    },
                    Object {
                      "label": "Pet Supplies",
                      "textColor": undefined,
                      "value": "4",
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Price *
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Price *"
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType="numeric"
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 17,
              "marginBottom": 2,
            }
          }
        >
          Pro tip: better to set affordable price
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Allow phone *
        </Text>
        <RCTSwitch
          accessibilityLabel="Allow phone *"
          disabled={false}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor={undefined}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Object {
                "marginBottom": 4,
              },
            ]
          }
          thumbTintColor={undefined}
          tintColor={undefined}
          value={false}
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          Phone *
        </Text>
        <TextInput
          accessibilityLabel="Phone *"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={false}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          options={Object {}}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#777777",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          type="cel-phone"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            }
          }
        >
          End time *
        </Text>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                }
              }
            >
              00:00
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginBottom": 4,
                  },
                  Object {
                    "height": 0,
                  },
                ]
              }
            >
              <RCTDatePicker
                date={949442400000}
                maximumDate={undefined}
                minimumDate={undefined}
                minuteInterval={undefined}
                mode="time"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 216,
                  }
                }
                timeZoneOffsetInMinutes={undefined}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add Product
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
